03-pipefilter.md

```diff
@@ -16,277 +16,465 @@ Now that we know a few basic commands,
-> the final '\*' can match any number of characters). Thus, `p*.p?*` would
-> match `preferred.practice`, and even `p.pi` (since the first '\*' can
-> match no characters at all), but not `quality.practice` (doesn't start
-> with 'p') or `preferred.p` (there isn't at least one character after the
-> '.p').
-> 
-> When the shell sees a wildcard, it expands the wildcard to create a
-> list of matching filenames *before* running the command that was
-> asked for. As an exception, if a wildcard expression does not match
-> any file, Bash will pass the expression as a parameter to the command
-> as it is. For example typing `ls *.pdf` in the molecules directory
-> (which contains only files with names ending with `.pdb`) results in
-> an error message that there is no file called `*.pdf`.
-> However, generally commands like `wc` and `ls` see the lists of
-> file names matching these expressions, but not the wildcards
-> themselves. It is the shell, not the other programs, that deals with
-> expanding wildcards, and this another example of orthogonal design.
+The greater than symbol, `>`, tells the shell to **redirect** the command's output
+to a file, in this case `content-of-molecules` instead of printing it to the screen.
+The shell will create the file if it doesn't exist,
+or overwrite the contents of that file if it does.
+(This is why there is no screen output:
+everything that `ls` would have printed has gone into the file
+`content-of-molecules` instead.)
+
+> ## Redirecting Input {.callout}
+>
+> As well as using `>` to redirect a program's output, we can use `<` to
+> redirect its input, i.e., to read from a file instead of from standard
+> input. For example, instead of writing `wc ammonia.pdb`, we could write
+> `wc < ammonia.pdb`. In the first case, `wc` gets a command line
+> parameter telling it what file to open. In the second, `wc` doesn't have
+> any command line parameters, so it reads from standard input, but we
+> have told the shell to send the contents of `ammonia.pdb` to `wc`'s
+> standard input.
 
-If we run `wc -l` instead of just `wc`,
-the output shows only the number of lines per file:
+For (2) we will use `wc`
+that is the "word count" command:
+it counts the number of lines, words, and characters in files.
+So,
 
 ~~~ {.bash}
-$ wc -l *.pdb
+$ wc content-of-molecules
 ~~~
 ~~~ {.output}
-  20  cubane.pdb
-  12  ethane.pdb
-   9  methane.pdb
-  30  octane.pdb
-  21  pentane.pdb
-  15  propane.pdb
- 107  total
+     48      48     645
 ~~~
 
-We can also use `-w` to get only the number of words,
-or `-c` to get only the number of characters.
-
-Which of these files is shortest?
-It's an easy question to answer when there are only six files,
-but what if there were 6000?
-Our first step toward a solution is to run the command:
+And to get only the number of files inside `molecules`
+we can run
 
 ~~~ {.bash}
-$ wc -l *.pdb > lengths.txt
+$ wc -l content-of-molecules
+~~~
+~~~ {.output}
+48
 ~~~
```

Comment:
Maybe add a concluding sentence like "we've answered our question!  There are 48 pdb files"